### PR TITLE
Feature/vsdspub 291 substring

### PR DESCRIPTION
**Pull request by WLefever-Cegeka**

### CODE REVIEW

File: ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+				ldesFragmentRepository);
+		SubstringFragmentFinder substringFragmentFinder = new SubstringFragmentFinder(substringFragmentCreator,
+				substringConfig, substringRelationsAttributer);
+		return new SubstringFragmentationStrategy(fragmentationStrategy,
+				ldesFragmentRepository,
+				tracer, substringBucketiser, substringFragmentFinder, substringFragmentCreator);
+	}
+
+	private SubstringConfig createSubstringConfig(FragmentationProperties properties) {
+		SubstringConfig geospatialConfig = new SubstringConfig();
```
Comment (sverholen) [inline ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java:38]: copy/paste :)

Comment (WLefever-Cegeka) [reply ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java:38]: Ha, well spotted :-)

**Review by sverholen — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -21,7 +21,7 @@ COPY --from=app-stage ldes-server-infra-mongo/target/ldes-server-infra-mongo-jar
 COPY --from=app-stage ldes-server-port-ingestion-rest/target/ldes-server-port-ingestion-rest-jar-with-dependencies.jar ./lib/
 COPY --from=app-stage ldes-server-port-fetch-rest/target/ldes-server-port-fetch-rest-jar-with-dependencies.jar ./lib/
 COPY --from=app-stage ldes-fragmentisers/ldes-fragmentisers-geospatial/target/ldes-fragmentisers-geospatial-jar-with-dependencies.jar ./lib/
-COPY --from=app-stage ldes-fragmentisers/ldes-fragmentisers-timebased/target/ldes-fragmentisers-timebased-jar-with-dependencies.jar ./lib/
+COPY --from=app-stage ldes-fragmentisers/ldes-fragmentisers-substring/target/ldes-fragmentisers-substring-jar-with-dependencies.jar ./lib/
```
Comment (Yalz) [inline Dockerfile:24]: Am i missing something or did we replace the existing timebased with substring?

File: ldes-queues/ldes-queue-in-memory/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/mediator/InMemoryFragmentationMediatorTest.java [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
+// {
+// Awaitility.reset();
+// LdesMember ldesMember = mock(LdesMember.class);
+// fragmentationMediator.addMemberToFragment(ldesMember);
+//
+// await().pollDelay(Durations.ONE_MILLISECOND)
+// .atMost(Durations.TEN_SECONDS)
+// .until(() -> fragmentationMediator.ldesMembersToFragment.isEmpty());
+//
+// verify(fragmentationExecutor, times(1)).executeFragmentation(ldesMember);
```
Comment (Yalz) [inline ldes-queues/ldes-queue-in-memory/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/mediator/InMemoryFragmentationMediatorTest.java:43]: We'll need to find a way to fix this test one day :)

**Review by Yalz — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (WLefever-Cegeka) [reply Dockerfile:25]: Oops!

**Pull request merged**